src/app/components/user-form/user-form/user-form.component.html

```diff
@@ -0,0 +1,65 @@
+<ng-container [formGroup]="userFormGroup">
+    <div>
+        <div>
+            <label for="firstName">First Name</label>
+            <input type="text" name="firstName" formControlName="firstName">
+            <span *ngIf="userFormGroup.get('firstName')?.errors?.['required']" class="error-message">Field is
+                required</span>
+        </div>
+        <div>
+            <label for="lastName">Last Name</label>
+            <input type="text" name="lastName" formControlName="lastName">
+            <span *ngIf="userFormGroup.get('lastName')?.errors?.['required']" class="error-message">Field is
+                required</span>
+        </div>
+        <div>
+            <label for="age">Age</label>
+            <input type="number" name="age" formControlName="age">
+            <span *ngIf="userFormGroup.get('age')?.errors?.['required']" class="error-message">Field is required</span>
+            <span *ngIf="userFormGroup.get('age')?.errors?.['min']" class="error-message">Age should be more than
+                5</span>
+            <span *ngIf="userFormGroup.get('age')?.errors?.['max']" class="error-message">Age should be less than
+                100</span>
+        </div>
+        <div>
+            <label for="company">Company</label>
+            <input type="text" name="company" formControlName="company">
+            <span *ngIf="userFormGroup.get('company')?.errors?.['maxLength']" class="error-message">The length should be
+                no more than 35</span>
+
+        </div>
+        <div>
+            <label for="email">Email</label>
+            <input type="text" name="email" formControlName="email">
+            <span *ngIf="userFormGroup.get('email')?.errors?.['required']" class="error-message">Field is
+                required</span>
+            <span *ngIf="userFormGroup.get('email')?.errors?.['email']" class="error-message">Email should be in correct
+                format </span>
+            <span *ngIf="userFormGroup.get('email')?.errors?.['gmailDomain']" class="error-message">Domain should be gmail.com</span>
+        </div>
+        <div>
+            <label for="department">Department</label>
+            <select id="department" formControlName="department">
+                <option *ngFor="let d of departments" [ngValue]="d">{{ d }}</option>
+            </select>
+        </div>
+        <div>
+            <label for="gender">Gender</label>
+            <div>
+                <input style="display: inline" id="female" type="radio" name="gender" value="F" formControlName="gender">
```

Comment:
Inline styles is a bad practice. It is more useful to place styles to separate SCSS/CSS file. If Angular Material is used in your project then mat-radio-group and mat-radio can be used.